cf/api/service_brokers.go

```diff
@@ -56,7 +56,7 @@ func (repo CloudControllerServiceBrokerRepository) FindByName(name string) (serv
 			return false
 		})
 
-	if !foundBroker {
+	if !foundBroker && (apiErr == nil) {
```

Comment:
Hmm, OK. I guess I'd then simplify the conditional into two. However, we are splitting hair here so if others are OK then I am happy to be cool with it. Cheers.